packages/core/src/tracing/instrumentation/database/db2.js

```diff
@@ -0,0 +1,612 @@
+  // CASE: db call is disabled via suppress header
+  return !isActive || cls.tracingSuppressed() || (ignoreClsTracing ? false : !cls.isTracing());
+}
+
+function instrumentOpen(originalFunction) {
+  return function instanaInstrumentationOpen() {
+    // NOTE: connection.open(fn) will throw an error in the library
+    //       we can rely on arguments[0] being the connection string.
+    //       There is no other format to pass in the connection.
+    connectionStr = arguments[0];
+    return originalFunction.apply(this, arguments);
+  };
+}
+
+function instrumentQueryResult(originalFunction) {
+  return function instanaInstrumentQueryResult(stmt) {
+    return instrumentQueryResultHelper(this, arguments, originalFunction, stmt, true);
+  };
+}
+
+function instrumentQueryResultSync(originalFunction) {
+  return function instanaInstrumentQueryResultSync(stmt) {
+    return instrumentQueryResultHelper(this, arguments, originalFunction, stmt, false);
+  };
+}
+
+/**
+ * We are losing the parentSpan because of the C++ implementation of `beginTransaction`.
+ * We were unable to figure out why the context get's lost.
+ *
+ * We thought it is related to not having used
+ * https://github.com/nodejs/nan/blob/main/doc/node_misc.md#api_nan_asyncresource
+ *
+ * That's why we have to instrument `beginTransaction` and reactivate
+ * the span.
+ *
+ * We do not loose the parentSpan if the customer uses "beginTransactionSync".
+ */
+function instrumentBeginTransaction(originalFunction) {
+  return function instanaInstrumentBeginTransaction() {
+    const parentSpan = cls.getCurrentSpan();
+    const originalCallback = arguments[0];
+
+    if (typeof originalCallback !== 'function') {
+      return originalCallback.apply(this, arguments);
+    }
+
+    arguments[0] = cls.ns.bind(function instanaInstrumentBeginTransactionCallback() {
+      // NOTE: See function description
+      cls.setCurrentSpan(parentSpan);
+      return originalCallback.apply(this, arguments);
+    });
+
+    return originalFunction.apply(this, arguments);
+  };
+}
+
+function instrumentQuery(originalFunction) {
+  return function (stmt) {
+    return instrumentQueryHelper(this, arguments, originalFunction, stmt, true);
+  };
+}
+
+function instrumentQuerySync(originalFunction) {
+  return function (stmt) {
+    return instrumentQueryHelper(this, arguments, originalFunction, stmt, false);
+  };
+}
+
+/**
+ * Main difference of prepare and a normal query call:
+ * - you can call prepare once and re-use the statement (higher performance)
+ */
+function instrumentPrepare(originalFunction) {
+  return function instanaInstrumentPrepare() {
+    const ctx = this;
+    const originalArgs = arguments;
+
+    // CASE: we pass "true" to skip `cls.isTracing` because the prepare call can happen on process start
+    if (skipTracing(true)) {
+      return originalFunction.apply(ctx, originalArgs);
+    }
+
+    const possibleHttpParentSpan = cls.getCurrentSpan();
```

Comment:
Discussed: 
We would like to add one more test case just to ensure we are doing everything right here:

- first http req (.prepare)
- second http req (.execute with the first http parent)

Reply:
> cls.getCurrentSpan()

Change to `cls.getCurrentEntrySpan()` to be 100% sure we are remembering the right span here.